distributed/tests/test_executor.py

```diff
@@ -3287,6 +3289,8 @@ def test_publish_bag(s, a, b):
     assert len(futures_of(bagp)) == 3
 
     result = yield f._get_dataset('data')
+    assert set(result.dask.keys()) == set(bagp.dask.keys())
+    assert [f.key for f in result.dask.values()] == [f.key for f in bagp.dask.values()]
```

Comment:
Recommend using curly braces here.  The output order of iterating over dictionaries is not guaranteed.